### ISSUE TITLE
Post Excerpt: Add design tools for "read more" link

### ISSUE THREAD

**Issue by mikachan** [WordPress/gutenberg#57352]
## What problem does this address?

The design tools available for the Post Excerpt block do not include separate styling for the "read more" link that is included in the block.

## What is your proposed solution?

Allow users to style this link separately in the UI. An example use case is being able to style this link like a button, with borders and a background.

**Comment by ntsekouras** [comment]
I think it's better to use the `Read More` block and leave the `read more` from the Post Excerpt block empty. This way you can style the link. I have a [very old PR](https://github.com/WordPress/gutenberg/pull/38662) that was trying to make this more intuitive but it's still there :)

**Comment by vipul0425** [comment]
@mikachan I have raised this [PR](https://github.com/WordPress/gutenberg/pull/57635/), Is there anything I can improve to get it merged?

Thanks.

**Comment by ntsekouras** [comment]
>@mikachan I have raised this https://github.com/WordPress/gutenberg/pull/57635/, Is there anything I can improve to get it merged?

Thanks for the PR @vipul0425 , but I still don't think we should proceed with this. We can already achieve the styling with `Read more` block and have way more flexibility, since we can move the block where ever we want and without the need of combined usage with Post Excerpt block.

--cc @WordPress/gutenberg-design

**Comment by vipul0425** [comment]
Alright! Thanks for the reply @ntsekouras. :)

**Comment by ntsekouras** [comment]
>Alright! Thanks for the reply @ntsekouras. :)

Let's wait for some more feedback from design 😄

**Comment by mikachan** [comment]
Thanks for looking into this @vipul0425 and creating a PR 🙏 It sounds like using the Read More block will provide the flexibility we need here, but let's see what the design feedback is.

**Comment by jasmussen** [comment]
If I understand this correctly, the desire is to add design options (typography, color, etc.) to the Read More _block_, and then per [Nik's comment here](https://github.com/WordPress/gutenberg/issues/57352#issuecomment-2090772992), to use that same Read More block inside the Post Excerpt block. Is that right?

If yes, then I'd support that. Styling Read More links comes up a lot in themes design.

**Comment by ntsekouras** [comment]
>If I understand this correctly, the desire is to add design options (typography, color, etc.) to the Read More block, 

No, The issue is to add design options inside Post Excerpt, for the read more link there which is not a standalone block. Post Excerpt block is a combination of two RichText instances. In order to bring another block inside, we would need to refactor Post Excerpt block to support inner blocks.

My suggestion is even if you use Post Excerpt block, **don't** use the RichText read more link there, but instead also add the standalone `Read more` block, which already has design options.

**Comment by jasmussen** [comment]
I'm confused, sorry 😅 — but it still sounds like we are in agreement. There should be just one way to style the "Read More" link, and ideally it's the same across.

**Comment by ntsekouras** [comment]
>I'm confused, sorry 😅 — but it still sounds like we are in agreement. There should be just one way to style the "Read More" link, and ideally it's the same across.

 I don't think we are on the same page 😄. I agree that ideally we should have only one way to have a `read more link`, but we don't. There are two ways:
1. The `Read more` standalone block whose flexibility (which also has design options) is something we should preserve for all the reasons I've mentioned above.
2. Inside Post excerpt block, as an optional RichText instance.

So in order to properly have one way of adding a `read more` link, we should update Post Excerpt block to have inner blocks and migrate the old ones. Yeah, in that case the issue could be changed to reflect this, but IMO, I don't think it's worth it. The only reason it was kept inside Post Excerpt when the standalone `Read more` block was implemented was for back compat.

**Comment by jasmussen** [comment]
Oh I see, you're suggesting to not use the Read More block as part of the post excerpt at all, but to manually insert the _block_ after the post excerpt block. Did I finally get it right? If yes, I agree. And I agree that the Read More block should get design tools as well.

**Comment by ntsekouras** [comment]
> but to manually insert the block after the post excerpt block. Did I finally get it right? 

Yes you did! Also noting that you can insert the `Read more` block wherever you want and not necessarily after Post excerpt block. 

>And I agree that the Read More block should get design tools as well.

The standalone `Read more` block already has design tools, so no further action is required to that front.

**Comment by jasmussen** [comment]
I wonder if we shouldn't then de-emphasize the read more option that's built-in to the excerpt:

![Screenshot 2024-05-03 at 11 50 02](https://github.com/WordPress/gutenberg/assets/1204802/9b48ab94-1b5f-4607-bc61-5d49a68e255d)

Can be a toggle defaulting to off for new blocks, perhaps? Can even have help text suggesting you use the standalone block for "more design options".

**Comment by ntsekouras** [comment]
>Can be a toggle defaulting to off for new blocks, perhaps? 

That's exactly what my two-years old [PR](https://github.com/WordPress/gutenberg/pull/38662) does 😄 . I have mentioned this PR in various issues, but there was never traction for this.. After all these years though, I have kept it open, because I still think that it might be merged at some point.

On the other hand, if there is no traction, it seems there might be no real problem with the current situation too..

**Comment by jasmussen** [comment]
It's been a tough couple of years, man! But good on keeping it open. Perhaps you can rebase and we can land it?
